app/domain/authentication/authn_gcp/validate_resource_restrictions_configuration.rb

```diff
@@ -0,0 +1,49 @@
+module Authentication
+  module AuthnGCP
+
+    # This class is responsible of validating resource restrictions combination.
+    # the allowed combination is: Role should have at least 1 permitted constraints, and non of illegal constraints
+    ValidateResourceRestrictionsConfiguration = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger
+      },
+      inputs:       %i(resource_restrictions permitted_constraints)
+    ) do
+
+      def call
+        @logger.debug(LogMessages::Authentication::AuthnGCP::ValidatingResourceRestrictionsConfiguration.new)
+        validate_restrictions_exist
+        validate_constraints_are_permitted
+        validate_restrictions_values_exist
+        @logger.debug(LogMessages::Authentication::AuthnGCP::ValidatedResourceRestrictionsConfiguration.new)
+      end
+
+      private
+
+      def validate_restrictions_exist
+        if @resource_restrictions.nil? || @resource_restrictions.empty?
+          raise Errors::Authentication::RoleMissingRequiredConstraints.new(@permitted_constraints)
+        end
+      end
```

Comment:
so here's something. when you read this code you see that if `@resource_restrictions` is empty then the role doesn't have the required constraints. But we don't see here any logic of reading the restrictions from the role. i am not saying that you should change the logic so that the validations happen in the extraction class but just wanted to point to this thing so you can consider it.

Reply:
Yes this is the downside of separating those logics, but i sill preferred it separated